In [ ]:
# import libraries
import pandas as pd
import sqlite3


In [2]:
# Load CSV
df = pd.read_csv(r"C:\Users\ACER\Desktop\Portfolios\Data Analysis\customer_churn\mtn_customer_churn.csv")

# Quick look at the data
df.head()


full_name date_of_purchase  age        state           mtn_device  \
0   Ngozi Berry        1/25/2025   27        Kwara            4G Router   
1  Zainab Baker        3/25/2025   16  Abuja (FCT)      Mobile SIM Card   
2   Saidu Evans        3/25/2025   21       Sokoto  5G Broadband Router   
3   Saidu Evans        3/25/2025   21       Sokoto      Mobile SIM Card   
4   Saidu Evans        3/25/2025   21       Sokoto       Broadband MiFi   

   gender  satisfaction_rate customer_review  customer_tenure_in_months  \
0    Male                  2            Fair                          2   
1  Female                  2            Fair                         22   
2    Male                  1            Poor                         60   
3    Male                  1            Poor                         60   
4    Male                  1            Poor                         60   

             subscription_plan  unit_price  number_of_time_purchased  \
0           165GB Monthly Plan       35000                        19   
1          12.5GB Monthly Plan        5500                        12   
2  150GB FUP Monthly Unlimited       20000                         8   
3       1GB+1.5mins Daily Plan         500                         8   
4  30GB Monthly Broadband Plan        9000                        15   

   total_revenue  data_usage customer_churn_status  \
0         665000       44.48                   Yes   
1          66000       19.79                   Yes   
2         160000        9.64                    No   
3           4000      197.05                    No   
4         135000       76.34                    No   

                reasons_for_churn  
0                      Relocation  
1  Better Offers from Competitors  
2                             NaN  
3                             NaN  
4                             NaN

In [3]:
# Check info
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 974 entries, 0 to 973
Data columns (total 16 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   full_name                  974 non-null    object 
 1   date_of_purchase           974 non-null    object 
 2   age                        974 non-null    int64  
 3   state                      974 non-null    object 
 4   mtn_device                 974 non-null    object 
 5   gender                     974 non-null    object 
 6   satisfaction_rate          974 non-null    int64  
 7   customer_review            974 non-null    object 
 8   customer_tenure_in_months  974 non-null    int64  
 9   subscription_plan          974 non-null    object 
 10  unit_price                 974 non-null    int64  
 11  number_of_time_purchased   974 non-null    int64  
 12  total_revenue              974 non-null    int64  
 13  data_usage                 974 non-null    float64

In [4]:
# Check missing values
df.isnull().sum()

full_name                      0
date_of_purchase               0
age                            0
state                          0
mtn_device                     0
gender                         0
satisfaction_rate              0
customer_review                0
customer_tenure_in_months      0
subscription_plan              0
unit_price                     0
number_of_time_purchased       0
total_revenue                  0
data_usage                     0
customer_churn_status          0
reasons_for_churn            690
dtype: int64

In [5]:
# Check for duplicates 
df.duplicated().sum()

np.int64(0)

In [6]:
# Descriptive statistics
df.describe()

age  satisfaction_rate  customer_tenure_in_months  \
count  974.000000         974.000000                 974.000000   
mean    48.043121           2.947639                  31.422998   
std     17.764307           1.384219                  17.191256   
min     16.000000           1.000000                   1.000000   
25%     32.000000           2.000000                  17.000000   
50%     49.000000           3.000000                  31.000000   
75%     63.750000           4.000000                  47.000000   
max     80.000000           5.000000                  60.000000   

          unit_price  number_of_time_purchased  total_revenue  data_usage  
count     974.000000                974.000000   9.740000e+02  974.000000  
mean    19196.663244                 10.564682   2.046696e+05   99.304764  
std     25586.726985                  5.709427   3.247855e+05   57.739511  
min       350.000000                  1.000000   3.500000e+02    0.820000  
25%      5500.000000                  5.000000   3.300000e+04   47.637500  
50%     14500.000000                 11.000000   1.080000e+05  103.330000  
75%     24000.000000                 15.000000   2.610000e+05  149.697500  
max    150000.000000                 20.000000   3.000000e+06  200.000000

* Age: 16–80 years, mean ~48; moderate variability, mostly middle-aged customers.

* Satisfaction: Avg 3/5; mixed satisfaction with room for improvement.

* Tenure: 1–60 months, median ~31; mix of new and long-term users.

* Unit Price & Purchases: Highly variable spending (350–150,000 NGN) and purchase frequency (1–20); few high spenders skew revenue.

* Total Revenue: Median 108,000 NGN, mean 204,670 NGN → revenue driven by top customers.

* Data Usage: Avg 99 GB, wide range (0.8–200 GB) → identifies light vs heavy users.

Insight: High-value and heavy data users stand out; average satisfaction suggests potential churn risk.

In [7]:
# Create a database

# Connect to SQLite (creates database if it doesn't exist)
conn = sqlite3.connect("mtn_churn.db")
cursor = conn.cursor()

In [8]:
# Load dataframe into SQLite
df.to_sql("customers", conn, if_exists="replace", index=False)


974

In [9]:
# Check if data was loaded
pd.read_sql("SELECT * FROM customers LIMIT 5;", conn)

full_name date_of_purchase  age        state           mtn_device  \
0   Ngozi Berry        1/25/2025   27        Kwara            4G Router   
1  Zainab Baker        3/25/2025   16  Abuja (FCT)      Mobile SIM Card   
2   Saidu Evans        3/25/2025   21       Sokoto  5G Broadband Router   
3   Saidu Evans        3/25/2025   21       Sokoto      Mobile SIM Card   
4   Saidu Evans        3/25/2025   21       Sokoto       Broadband MiFi   

   gender  satisfaction_rate customer_review  customer_tenure_in_months  \
0    Male                  2            Fair                          2   
1  Female                  2            Fair                         22   
2    Male                  1            Poor                         60   
3    Male                  1            Poor                         60   
4    Male                  1            Poor                         60   

             subscription_plan  unit_price  number_of_time_purchased  \
0           165GB Monthly Plan       35000                        19   
1          12.5GB Monthly Plan        5500                        12   
2  150GB FUP Monthly Unlimited       20000                         8   
3       1GB+1.5mins Daily Plan         500                         8   
4  30GB Monthly Broadband Plan        9000                        15   

   total_revenue  data_usage customer_churn_status  \
0         665000       44.48                   Yes   
1          66000       19.79                   Yes   
2         160000        9.64                    No   
3           4000      197.05                    No   
4         135000       76.34                    No   

                reasons_for_churn  
0                      Relocation  
1  Better Offers from Competitors  
2                            None  
3                            None  
4                            None

- CSV extracted, cleaned, and loaded into SQLite.  
- You can now query the `customers` table for analysis.
